docs/man/awslocal.1.md
---
title: awslocal(1)
section: 1
version: 1.0.1
builddate: 1.0.1
date: $builddate$
header: awslocal AwsMock local command
footer: awslocal $version$
---

## NAME

```awslocal``` AwsMock local command.

## DESCRIPTION

```awslocal``` is a short bash script, which can be used to simplify the command line execution of AWS commands against
the AwsMock AWS simulation. It sets the endpoint and the profile to be used for the AwsMock command line commands.
Credentials are normally not used during the work with AwsMock. Nevertheless, you can give a AWS profile name in the
bash script in order to provide AWS login credentials.

```awslocal``` is per default installed in ```/usr/bin``` (if you
use the docker image, it will be in ```/usr/bin``` inside the docker image). If you need it locally, copy it from the
docker image.

## EXAMPLES

To get a list of all currently available queues:

```
awslocal sqs list-queues
{
    "QueueUrls": [
        "http://localhost:4566/000000000000/test-dlqueue",
        "http://localhost:4566/000000000000/test-queue-queue",
        "http://localhost:4566/000000000000/test-result-dlqueue",
        "http://localhost:4566/000000000000/test-result-queue",
        ...
}
```

To get a list of all SQS queue attributes:

```
awslocal sqs get-queue-attributes --queue-url http://localhost:4566/000000000000/test-queue --attribute-name All
{
    "Attributes": {
        "ApproximateNumberOfMessages": "0",
        "ApproximateNumberOfMessagesDelayed": "0",
        "ApproximateNumberOfMessagesNotVisible": "0",
        "CreatedTimestamp": "1702377771",
        "DelaySeconds": "0",
        "LastModifiedTimestamp": "1702377771",
        "MaximumMessageSize": "262144",
        "MessageRetentionPeriod": "345600",
        "Policy": "",
        "QueueArn": "arn:aws:sqs:eu-central-1:000000000000:test-queue",
        "ReceiveMessageWaitTimeSeconds": "20",
        "VisibilityTimeout": "30"
    }
}
```

## AUTHOR

Jens Vogt <jens.vogt@opitz-consulting.com>

## VERSION

$version$ ($builddate$)

## BUGS

Bugs and enhancement requests can be reported and filed at https://github.com/jensvogt/awsmock/issues

## SEE ALSO

```awsmockctl(1)```, ```awsmockmgr(1)```, ```awsmocksqs(1)```, ```awsmocks3(1)```, ```awsmocksns(1)```,
```awsmocklambda(1)```, ```awsmockdynamodb(1)```, ```awsmockcognito(1)```, ```awsmocktransfer(1)```,
```awsmocksecretsmanager(1)```, ```awsmocksqs(1)```, ```awsmockssm(1)```, ```awsmockapi(1)```

CHANGELOG.md
# Changelog

## [1.7.0](https://github.com/jensvogt/awsmock/compare/v1.6.4...v1.7.0) (2025-09-10)


### Features

* **123-windows-port:** windows port ([a48d876](https://github.com/jensvogt/awsmock/commit/a48d87647b3928def7a356c2027a7508c53b4020))


### Bug Fixes

* **1375-lambda-invocation:** add enabled flag to lambdas ([25480bb](https://github.com/jensvogt/awsmock/commit/25480bb53cbee84cc7aefa038904a6a99e91372c))
* **1375-lambda-invocation:** add start/stop/restart/enable/disable applications from awsmokctl ([f68f16f](https://github.com/jensvogt/awsmock/commit/f68f16fc4f60c2844686a748895dfaf0f64f8621))
* **1375-lambda-invocation:** add start/stop/restart/enable/disable applications from awsmokctl ([f3f6353](https://github.com/jensvogt/awsmock/commit/f3f63536ebcb591247ad2cf6ab6dee9a204ef6e0))
* **1375-lambda-invocation:** add start/stop/restart/enable/disable lambda to awsmockctl ([b2d09dd](https://github.com/jensvogt/awsmock/commit/b2d09ddb849d47e56db98082e1aa7d4a96bff75d))
* **1375-lambda-invocation:** add start/stop/restart/enable/disable lambda to awsmockctl ([eb30e26](https://github.com/jensvogt/awsmock/commit/eb30e269d0c98f718429179551fbebf07d997587))
* **1375-lambda-invocation:** cleanup code ([768faf6](https://github.com/jensvogt/awsmock/commit/768faf6f39fb2ab7a719beccab468d5c69c01b79))
* **1375-lambda-invocation:** cleanup code ([80fe7e7](https://github.com/jensvogt/awsmock/commit/80fe7e7c698e42496af535ba5b84191a8bdaa322))
* **1375-lambda-invocation:** cleanup code ([7a853ec](https://github.com/jensvogt/awsmock/commit/7a853ece974f6618d0ee1197c73538a19b0814c3))
* **1375-lambda-invocation:** cleanup code ([4486f8a](https://github.com/jensvogt/awsmock/commit/4486f8a44b8cb9bdb33440f0df19191278ba1cc8))
* **1375-lambda-invocation:** fix angular version ([8884bbd](https://github.com/jensvogt/awsmock/commit/8884bbd59a93748deefbc25df63c5ef4f3171639))
* **1375-lambda-invocation:** fix lambda enabled setting ([d816f7d](https://github.com/jensvogt/awsmock/commit/d816f7dc37e84754d31ddc4b5ba1d96839e9ce25))
* **1375-lambda-invocation:** fix lambda startup status ([b40aa8a](https://github.com/jensvogt/awsmock/commit/b40aa8aab515eecf583180c80e3ad392dc84bb20))
* **1375-lambda-invocation:** fix memory leak ([072acc7](https://github.com/jensvogt/awsmock/commit/072acc720714a9928ab18252c2f084180990eb25))
* **1375-lambda-invocation:** fix memory leak ([0f53122](https://github.com/jensvogt/awsmock/commit/0f531221a87aae1bc29341dd9bf469f9d4624e47))
* **1375-lambda-invocation:** fix memory leak ([7172f2d](https://github.com/jensvogt/awsmock/commit/7172f2dc41d618245dfe2102ad8146d4b2904b5f))
* **1375-lambda-invocation:** fix tests ([3d114d1](https://github.com/jensvogt/awsmock/commit/3d114d1881e22fdefff4f6216ea1e1d4b0712dd5))
* **1375-lambda-invocation:** general cleanup ([0c36c1b](https://github.com/jensvogt/awsmock/commit/0c36c1bb90d437853c2500f6f89dc8564a27acfd))
* **1375-lambda-invocation:** general cleanup ([b6466fb](https://github.com/jensvogt/awsmock/commit/b6466fb2abc70a11c30555bd7d34e176950ce68d))
* **1375-lambda-invocation:** use boost ioc for detached threads ([a625f85](https://github.com/jensvogt/awsmock/commit/a625f85dba4492ad872cefcf62d3c8a8efacdba3))
* **1375-lambda-invocation:** use boost ioc for detached threads ([084ed92](https://github.com/jensvogt/awsmock/commit/084ed92b0630d4d33078b5481bfd3279592c892b))
* **1375-lambda-invocation:** use boost ioc for detached threads ([ab3471b](https://github.com/jensvogt/awsmock/commit/ab3471bf0ddf4bd9ce7ed5cac88836e91249529c))
* **1375-new-stream-response:** fix duplicate lambda event sources ([10c613a](https://github.com/jensvogt/awsmock/commit/10c613a33b1177184a391c0b0a1c4468e5643ecb))
* **1375-new-stream-response:** fix duplicate lambda event sources ([4caf485](https://github.com/jensvogt/awsmock/commit/4caf48542f9ecc81f3997233769efbab67f95a15))
* **1392-monitoring-when-service-is-down:** add first api gateway REST APIs ([6edc225](https://github.com/jensvogt/awsmock/commit/6edc225a61e448432a6adf984bf966f894c95f45))
* **1392-monitoring-when-service-is-down:** add first api gateway REST APIs ([04da21f](https://github.com/jensvogt/awsmock/commit/04da21fe017f6fb1f8cf3caea22e73a9cb51ad15))
* **1392-monitoring-when-service-is-down:** fix asynchron handler methods ([c47351d](https://github.com/jensvogt/awsmock/commit/c47351d6a980b0aca38d9950ce9046fff0bd66b5))
* **1392-monitoring-when-service-is-down:** fix asynchron handler methods ([ea3351d](https://github.com/jensvogt/awsmock/commit/ea3351db258cbdf4ee93d3a461618bf915d7535c))
* **1392-monitoring-when-service-is-down:** fix asynchron handler methods ([2ad5eec](https://github.com/jensvogt/awsmock/commit/2ad5eecc149c6c3b6418ffc207a7ccb152aae5c8))
* **1392-monitoring-when-service-is-down:** fix asynchron handler methods ([ce77b96](https://github.com/jensvogt/awsmock/commit/ce77b9638866788c8622acf9882a35cd92d0db22))
* **1392-monitoring-when-service-is-down:** fix asynchron handler methods ([a975025](https://github.com/jensvogt/awsmock/commit/a9750259d37c738c89d4e51114fb05e5a5c34940))
* **1392-monitoring-when-service-is-down:** fix asynchron handler methods ([4e4047d](https://github.com/jensvogt/awsmock/commit/4e4047db2d0f7f32cbbf6b9ceadd8efbafbfc838))
* **1392-monitoring-when-service-is-down:** fix asynchron handler methods ([6d8c2c8](https://github.com/jensvogt/awsmock/commit/6d8c2c8179526df6d0a03a65d59b3bf0760ad88f))
* **1392-monitoring-when-service-is-down:** fix asynchron handler methods ([4c903d4](https://github.com/jensvogt/awsmock/commit/4c903d45b955be4e8e1a650666f99ba1b66f36aa))
* **1392-monitoring-when-service-is-down:** fix asynchron handler methods ([2fa77ad](https://github.com/jensvogt/awsmock/commit/2fa77ad02076ef38efba8c48f57f2de32ac9149e))
* **1392-monitoring-when-service-is-down:** fix asynchron handler methods ([b1a5c08](https://github.com/jensvogt/awsmock/commit/b1a5c08b23ce56180fe4369cb12fd0549b3dd3dd))

src/awslocal/src/AwsLocal.cpp
//
// Created by vogje01 on 10/23/23.
//

#include <awsmock/awslocal/AwsLocal.h>

namespace AwsMock::AwsLocal {

    void AwsLocal::Initialize(const boost::program_options::variables_map &vm) {
        if (vm.contains("host")) {
            _host = vm.at("host").as<std::string>();
            _baseUrl = "http://" + _host + ":" + std::to_string(_port);
        }
        if (vm.contains("port")) {
            _port = vm.at("port").as<int>();
            _baseUrl = "http://" + _host + ":" + std::to_string(_port);
        }
        if (vm.contains("profile")) {
            _profile = vm.at("profile").as<std::string>();
            ReadAwsConfigFile();
        }
    }


    void AwsLocal::Run(std::vector<std::string> &command) const {
        std::string out, err;
        command.emplace_back("--endpoint");
        command.emplace_back(_baseUrl);
        Core::SystemUtils::RunShellCommand(AWS_CMD, command, out, err);
        if (!out.empty()) {
            std::cout << out;
        }
        if (!err.empty()) {
            std::cerr << err;
        }
    }

    void AwsLocal::ReadAwsConfigFile() {

        const std::string fileName = Core::SystemUtils::GetHomeDir() + "/.aws/config";
        if (!Core::FileUtils::FileExists(fileName)) {
            log_error << "In order to use profiles, you need to provide a ~/.aws/config file";
            return;
        }
        boost::property_tree::ptree pt;
        read_ini(fileName, pt);
        if (pt.get_optional<std::string>(_profile + ".endpoint_url").is_initialized()) {
            _baseUrl = pt.get<std::string>(_profile + ".endpoint_url");
        }
    }
}// namespace AwsMock::AwsLocal